packages/h5wasm/src/utils.ts

```diff
@@ -43,6 +46,14 @@ export function convertMetadataToDType(metadata: Metadata): DType {
     };
   }
 
+  throw new Error('Expected numeric metadata');
+}
+
+export function convertMetadataToDType(metadata: Metadata): DType {
```

Comment:
The addition of an another `if` clause trigged a cognitive complexity warning from sonarjs.

I therefore reworked `convertMetadataToDType` to separate the treatement of numerical dtypes in another `convertNumericMetadataToDType` function. This allows to remove the override signatures of `convertMetadataToDType`.